update select_mock_targets to use v3.0 basis templates

moustakas opened the pull request:
This PR is a companion to https://github.com/desihub/desisim/pull/434 and requires the new (`v3.0`) basis templates.

It also fixes the issue identified in #384 (using extinction-corrected colors) as well as improves the Gaussian mixture models used by `select_mock_targets` to assign more representative/accurate colors, shapes, and sizes to mock targets.

In addition, I addressed (finally!) yet another issue which was resulting in lower-than expected target densities for several target classes.

Finally, there is much better spectral coverage of the simulated LRG spectra (thanks to https://github.com/desihub/desisim/pull/433), and proper differentiation of north/south photometry for all target classes.

I have a couple more checks to do but this PR should be ready to be merged later today.

sbailey commented:
testing with this command:
```
srun -A desi -N 15 -n 30 -c 16 -C haswell -t 01:00:00 --qos interactive mpi_select_mock_targets \
--output_dir /global/cscratch1/sd/sjbailey/desi/dev/end2end/targets \
--config /global/cscratch1/sd/sjbailey/desi/dev/end2end/targets/select-mock-targets.yaml \
--seed 1 --nproc 16 --nside 64 \
--tiles /global/cscratch1/sd/sjbailey/desi/dev/end2end/targets/test-tiles.fits
```
I get errors like this:
```
INFO:parallel.py:412:stdouterr_redirected: End log redirection to /global/cscratch1/sd/sjbailey/desi/dev/end2end/targets/52/5274/build-64-5274.log at Wed Oct 10 09:58:03 2018
ERROR:mpi_select_mock_targets:176:<module>: Pixels [5274] failed after 9.6 minutes
ERROR:mpi_select_mock_targets:179:<module>: Traceback (most recent call last):
  File "/global/cscratch1/sd/sjbailey/desi/code/desitarget/bin/mpi_select_mock_targets", line 170, in <module>
    healpixels=rankpix[i:i+n], no_spectra=args.no_spectra)
  File "/global/cscratch1/sd/sjbailey/desi/code/desitarget/py/desitarget/mock/build.py", line 634, in targets_truth
    MakeMock=AllMakeMock[ii])
  File "/global/cscratch1/sd/sjbailey/desi/code/desitarget/py/desitarget/mock/build.py", line 173, in read_mock
    nside_galaxia=nside_galaxia, mock_density=mock_density)
  File "/global/cscratch1/sd/sjbailey/desi/code/desitarget/py/desitarget/mock/mockmaker.py", line 4696, in read
    mock_density=mock_density)
TypeError: readmock() got an unexpected keyword argument 'only_coords'
```

moustakas commented:
Thanks, yeah, that was an issue with the skies that I debugged this morning.  Please `git pull` this branch and retest.  I ran a number of healpixels to successful completion.  

Also note that you'll want to use/update your configuration file to what's in `desitarget/doc/mock_example/input.yaml`.

moustakas commented:
Thanks for merging, @sbailey.

Some other quick notes about this PR, so others are aware:

* I updated the default (all-sky) white-dwarf mock to be `v1.0.0` (from `v0.0.2`) from Boris, which includes mocked-up Gaia columns.  Consequently, in the code we now select `MWS_WD` and `STD_WD` targets "correctly" (i.e., using the official cuts in `desitarget.cuts`).  See [here](https://github.com/desihub/desitarget/issues/390#issuecomment-427111396) for details.

* I updated the default `ELG`, `QSO`, and `LRG` mocks from `v0.0.8_LPT` to `DarkSky/v1.0.1`, at the behest of @michaelJwilson (see [here](https://github.com/desihub/desitarget/issues/390#issuecomment-424874205)).  Although the DarkSky mocks aren't all-sky, they do cover a fairly conservative portion of the sky (all the way down to -30 deg Dec).

* I updated the default `LYA` mock from `london/v2.0` to `london/v4.0`.  See the README in `/global/projecta/projectdirs/desi/mocks/lya_forest/london/v4.0` for details about the improvements.

But still no contaminants...
